source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -362,13 +362,19 @@ void Router::onUpstreamData(Buffer::Instance& data, bool end_stream) {
       case MessageType::Reply:
         incClusterScopeCounter({upstream_resp_reply_});
         if (callbacks_->responseSuccess()) {
+          upstream_request_->upstream_host_->outlierDetector().putResult(
+              Upstream::Outlier::Result::ExtOriginRequestSuccess);
           incClusterScopeCounter({upstream_resp_reply_success_});
         } else {
+          upstream_request_->upstream_host_->outlierDetector().putResult(
+              Upstream::Outlier::Result::ExtOriginRequestFailed);
           incClusterScopeCounter({upstream_resp_reply_error_});
         }
         break;
 
       case MessageType::Exception:
+        upstream_request_->upstream_host_->outlierDetector().putResult(
+            Upstream::Outlier::Result::ExtOriginRequestFailed);
```

Comment:
Should we handle the reset and end stream cases below? They imply something went wrong decoding the response.

Reply:
+1

Reply:
I've added reporting to `onResetStream`, which should handle the `end_stream` case. I'm not sure what result should be reported in the case of reset, does `ExtOriginRequestFailed` make sense?

Reply:
I think so. It's an early termination of a response.

Reply:
ok，I've added code to handle this